packages/predicates/src/plugins/ownership-predicate.ts

```diff
@@ -0,0 +1,101 @@
+import BigNum = require('bn.js')
+
+import {
+  getOverlappingRange,
+  PredicatePlugin,
+  isRangeSubset,
+  StateUpdate,
+  SyncManager,
+  Transaction,
+} from '@pigi/core'
+
+export class OwnershipPredicatePlugin implements PredicatePlugin {
+  private readonly syncManager: SyncManager
+
+  public constructor(syncManager: SyncManager) {
+    this.syncManager = syncManager
+  }
+
+  public async executeStateTransition(
+    previousStateUpdate: StateUpdate,
+    transaction: Transaction,
+    witness: string
+  ): Promise<StateUpdate> {
+    await this.validateStateTransition(
+      previousStateUpdate,
+      transaction,
+      witness
+    )
+
+    return {
+      range: transaction.range,
+      stateObject: transaction.parameters.newState,
+      depositAddress: transaction.depositAddress,
+      plasmaBlockNumber: transaction.parameters.originBlock,
+    }
+  }
+
+  /**
+   * Validates that the provided previous StateUpdate, Transaction, and witness are valid.
+   *
+   * @param previousStateUpdate the previous StateUpdate upon which the provided Transaction acts
+   * @param transaction the Transaction to execute
+   * @param witness the signature data for the transaction
+   *
+   * @throws if the state transition is not valid or input is not of expected format
+   */
+  private async validateStateTransition(
+    previousStateUpdate: StateUpdate,
+    transaction: Transaction,
+    witness: string
+  ): Promise<void> {
+    // TODO: Actually check signature stuffs
+    if (previousStateUpdate.stateObject.data.owner !== witness) {
+      throw new Error(
+        `Cannot transition from state [${JSON.stringify(
+          previousStateUpdate
+        )}] with transaction [${JSON.stringify(
+          transaction
+        )}] because witness does not match previousStateUpdate owner.`
+      )
+    }
+
+    if (
+      previousStateUpdate.plasmaBlockNumber.gte(
+        transaction.parameters.originBlock
+      )
+    ) {
+      throw new Error(
+        `Cannot transition from state [${JSON.stringify(
+          previousStateUpdate
+        )}] with transaction [${JSON.stringify(
+          transaction
+        )}] because block number [${transaction.parameters.originBlock.toNumber()}] is not greater than previous state block number.`
+      )
+    }
+
+    if (!isRangeSubset(transaction.range, previousStateUpdate.range)) {
+      throw new Error(
+        `Cannot transition from state [${JSON.stringify(
+          previousStateUpdate
+        )}] with transaction [${JSON.stringify(
+          transaction
+        )}] because transaction range is not a subset of previous state update range.`
+      )
+    }
+
+    const lastSyncedBlock: BigNum = await this.syncManager.getLastSyncedBlock(
+      previousStateUpdate.stateObject.predicateAddress
+    )
+
+    if (lastSyncedBlock.gte(transaction.parameters.targetBlock)) {
```

Comment:
Ah shoot--we've just run into an issue where my irresponsible PR didn't update the docs to reflect changes in the contract--my bad!  We discovered in implementing the contract that ``targetBlock`` isn't needed because the deprecation call never actually checks where the block is included.  However, this still is worth commenting on, since if we did need ``targetBlock``, it would have to be done a little differently.  

The sync manager is meant to be a coordinator for the state manager, but calls within the state manager itself should not use it--for example, if we had previously synced block 100, and then received a new coin, we would need to use this function to verify history in blocks 1-99 and not care about having already synced to a different block.  In fact, with the new model for message queue/PubNub stuff discussed with Matt yesterday, we may not need the sync manager at all, and instead the message queue will automatically tell us exactly what to "sync" based on any incoming messages which involve us e.g. receiving new ranges.

I know that's not quite the check you're doing here--if it's greater than the last synced block, clearly we're skipping history and doing something wrong.  Luckily, that check is automatically handled by the ``verifiedBlockNumber`` check in ``executeTransaction`` itself.  Basically, since we end up verifying different block heights for different ranges, any notion of a globally synced block is irrelevant for a particular transaction verification--we *always* need to check that we have verified up to the last block before accepting a transaction, and so this is handled always via ``verifiedStateUpdates`` in the block manager, and we can ignore it for predicates and just deal with the transition logic. :)